UI: E2E testing of Solutions UI integration

carlito-scality opened the pull request:
**Component**:UI/Cypress tests

<!-- E.g. 'salt', 'containers', 'kubernetes', 'build', 'tests'... -->

**Context**: 
- Implement a Cypress test suite for Environment creation and preparation from Platform UI  to prepare future integration to CI
- This test can only be tested on local. It is skipped in the CI

- Wait for https://github.com/scality/metalk8s/issues/1850

**Summary**:

**Acceptance criteria**: 

To test:

- Launch the UI on local with a solution "example-solution" imported
- Go to the project `ui` folder
- Run `./node_modules/cypress/bin/cypress run --spec 'cypress/integration/CreateEnvironment.feature'`

---

<!-- Declare one or more issues to close once this PR gets merged -->

Closes: https://github.com/scality/metalk8s/issues/1851

<!-- If you want to refer to an issue while not closing it, use:

See: #ISSUE_NUMBER

-->

carlito-scality committed:
UI: Fix- Refresh environments only in the Solutions page
- The environments are currently refreshed everywhere in the UI. Howerver, it should be only refresh when we are in the Solutions page

Fixs: #1851

carlito-scality committed:
UI: Add Environment creation Cypress tests
- This test checks the environment creation form works well

Fixs: #1851

bert-e commented:
# Hello carlito-scality,



My role is to assist you with the merge of this
pull request. Please type `@bert-e help` to get information
on this process, or consult the [user documentation](
https://eve.devsca.com/github/scality/metalk8s/bert-e/doc/user).





*Status report is not available.*

bert-e commented:
# Waiting for approval



The following approvals are needed before I can proceed with the merge:


* the author


* one peer





Peer approvals *must* include at least 1 approval from the following list:

* @cmonfort

* @alexandre-allard-scality

* @sayf-eddine-scality

* @thomasdanan

* @carlito-scality

* @teddyandrieux

* @nootal

* @lucieleonard

* @gdemonet

* @jbertran

* @nicolast

* @monpote

* @tcarmet

* @slaperche-scality

carlito-scality committed:
UI: Add "Add solution to an Environment" cypress tests
- This test check if we go to the correct solution UI route to prepare the environment

Fixs: #1851

bert-e commented:
# Waiting for approval



The following approvals are needed before I can proceed with the merge:


* the author


* one peer





Peer approvals *must* include at least 1 approval from the following list:

* @tcarmet

* @jbertran

* @carlito-scality

* @slaperche-scality

* @nicolast

* @teddyandrieux

* @monpote

* @sayf-eddine-scality

* @nootal

* @alexandre-allard-scality

* @gdemonet

* @lucieleonard

* @thomasdanan

* @cmonfort

carlito-scality committed:
Merge branch 'development/2.4' into feature/add-solution-environment-cypress-test

* development/2.4: (38 commits)
  ui/cypress Update Cypress binary and package link
  ...

carlito-scality committed:
Solution UI: Add solution existence check
- We have to make sur that the solutio is already imported before preparing the environment
Fixs: #1851

carlito-scality committed:
Skip CreateEnvironment.feature in CI tests
- CreateEnvironment.feature can only tested in local for now.
- This commit should be reverted once all ressources needed for CreateEnvironment.feature are available in the CI

Fixs: #1851

bert-e commented:
# Branches have diverged



This pull request's source branch `feature/add-solution-environment-cypress-test` has diverged from
`development/2.4` by more than 50 commits.

To avoid any integration risks, please re-synchronize them using one of the
following solutions:

* Merge `origin/development/2.4` into `feature/add-solution-environment-cypress-test`
* Rebase `feature/add-solution-environment-cypress-test` onto `origin/development/2.4`

Note: If you choose to rebase, you may have to ask me to rebuild
integration branches using the `reset` command.

bert-e commented:
# Waiting for approval



The following approvals are needed before I can proceed with the merge:


* the author


* one peer





Peer approvals *must* include at least 1 approval from the following list:

* @tcarmet

* @jbertran

* @carlito-scality

* @slaperche-scality

* @nicolast

* @teddyandrieux

* @monpote

* @sayf-eddine-scality

* @nootal

* @alexandre-allard-scality

* @gdemonet

* @lucieleonard

* @thomasdanan

* @cmonfort

carlito-scality commented:
@bert-e reset

bert-e commented:
# Reset complete



I have successfully deleted this pull request's integration branches.

bert-e commented:
# Waiting for approval



The following approvals are needed before I can proceed with the merge:


* the author


* one peer





Peer approvals *must* include at least 1 approval from the following list:

* @tcarmet

* @jbertran

* @carlito-scality

* @slaperche-scality

* @nicolast

* @teddyandrieux

* @monpote

* @sayf-eddine-scality

* @nootal

* @alexandre-allard-scality

* @gdemonet

* @lucieleonard

* @thomasdanan

* @cmonfort

carlito-scality committed:
Merge branch 'development/2.4' into feature/add-solution-environment-cypress-test

* development/2.4: (13 commits)
  vagrant/ubuntu: Add a repo update on Ubuntu VM
  ...

gdemonet commented:
Done in #2341
